## Data Cleaning - HOTEL 1

## Import the libraries

In [4]:
import pandas as pd
import plotly.express as px
import numpy as np
import datetime

## Import the data

### 11 tables of Hotel_1

#### 1 - Upsell tábla
- 83 sor
- adattípusok OK
- nincs NaN,
- gond: EUR és HUF összegek egy oszlopban vannak
- gond: vannak teljesen egyforma sorok, de ez életszerű, kérhetett ugyanaz a foglalás többször ugyanolyan szolgáltatást pl 71, 72, 73. sorok

In [16]:
upsell = pd.read_csv('./../Morgens_notebooks/HOTEL_1/Original_data/upsell_data_hotel_1.csv', sep=';')
upsell.head(3)

search_log_id  upsell_type        name  unit_price  pieces  sum_price
0         112857            7     ROSE_HU      3500.0       1     3500.0
1         114033           10  NONABAR_HU      4800.0       1     4800.0
2         114367            6   FRUITS_HU      3500.0       1     3500.0

In [7]:
upsell[upsell['search_log_id'] == 112758]

Empty DataFrame
Columns: [search_log_id, upsell_type, name, unit_price, pieces, sum_price]
Index: []

In [8]:
upsell.describe()

search_log_id  upsell_type    unit_price     pieces     sum_price
count      83.000000    83.000000     83.000000  83.000000     83.000000
mean   130854.590361    10.180723   5837.964940   1.265060   7770.180361
std      9769.080974     7.293740   3817.431666   0.495943   6998.472509
min    112857.000000     1.000000      9.700000   1.000000      9.700000
25%    120948.500000     5.000000   3500.000000   1.000000   3500.000000
50%    131444.000000     9.000000   5950.000000   1.000000   6000.000000
75%    138516.000000    16.000000   8350.000000   1.000000   9450.000000
max    146067.000000    37.000000  14200.000000   3.000000  26400.000000

In [9]:
upsell['name'].count()

83

In [10]:
upsell.groupby('name').count().head(3)

search_log_id  upsell_type  unit_price  pieces  sum_price
name                                                                    
CAKE_HU                    8            8           8       8          8
CAPPUCCINO_DE              5            5           5       5          5
CAPPUCCINO_HU              2            2           2       2          2

In [11]:
#54 db foglaláshoz tartozik a 83 sor, a search_log_id itt nem egyedi azonosító
len(upsell['search_log_id'].unique())

54

In [12]:
#no duplicated rows
upsell[upsell.duplicated()]

search_log_id  upsell_type       name  unit_price  pieces  sum_price
30         129168            1  DAYSPA_DE        23.2       2       46.4
72         141565            1  DAYSPA_HU      8350.0       1     8350.0
73         141565            1  DAYSPA_HU      8350.0       1     8350.0

In [13]:
upsell.to_csv('./../Morgens_notebooks/HOTEL_1/Clean_data/upsell_hotel_1.csv')

#### 2 - Booking_data
-802 sor
- adattípusok rendben
- gond: EUR és HUF összegek egy oszlopban vannak

In [18]:
booking = pd.read_csv('./../Morgens_notebooks/HOTEL_1/Original_data//booking_data_hotel_1.csv', sep=';')

In [20]:
booking.head(3)

search_log_id  total_price_final  rooms_total_price  upsell_total_price  \
0         112857           99683.00          119952.22              3500.0   
1         112906          302444.00          377114.44                 0.0   
2         113050             622.43             793.16                 0.0   

   vouchers_total_price  loyalty_discount_total  redeemed_loyalty_points_total  
0                     0                11995.22                        11774.0  
1                     0                37711.44                        36959.0  
2                     0                  118.97                            0.0

In [17]:
booking[booking['search_log_id'] == 112645]

Empty DataFrame
Columns: [search_log_id, total_price_final, rooms_total_price, upsell_total_price, vouchers_total_price, loyalty_discount_total, redeemed_loyalty_points_total]
Index: []

In [18]:
booking.describe()

search_log_id  total_price_final  rooms_total_price  \
count     802.000000       8.020000e+02       8.020000e+02   
mean   131071.062344       1.582970e+05       1.804741e+05   
std      9462.501514       1.421132e+05       1.572468e+05   
min    112857.000000       1.776800e+02       1.836000e+02   
25%    123174.500000       7.497500e+04       8.870444e+04   
50%    133948.000000       1.375850e+05       1.575972e+05   
75%    138939.250000       2.233800e+05       2.514491e+05   
max    146108.000000       1.192546e+06       1.281987e+06   

       upsell_total_price  vouchers_total_price  loyalty_discount_total  \
count          802.000000            802.000000              802.000000   
mean           804.145848            249.376559            17362.110224   
std           4056.736084           4990.643516            17952.527053   
min              0.000000              0.000000                0.000000   
25%              0.000000              0.000000               83.670000   
50%              0.000000              0.000000            14787.110000   
75%              0.000000              0.000000            25644.247500   
max          56050.000000         100000.000000           135749.290000   

       redeemed_loyalty_points_total  
count                     802.000000  
mean                     5188.450100  
std                     11136.605166  
min                         0.000000  
25%                         0.000000  
50%                         0.000000  
75%                      5000.000000  
max                    108953.000000

In [19]:
#no duplicated rows
booking[booking.duplicated()]

Empty DataFrame
Columns: [search_log_id, total_price_final, rooms_total_price, upsell_total_price, vouchers_total_price, loyalty_discount_total, redeemed_loyalty_points_total]
Index: []

In [20]:
len(booking['search_log_id'].unique())

802

In [21]:
booking.to_csv('./../Morgens_notebooks/HOTEL_1/Clean_data/booking_hotel_1.csv')

#### 3 - Child tábla (search_log_room_child)
- 14025 sor
-  adattípusok OK
-  nincs NaN

In [22]:
child = pd.read_csv('./../Morgens_notebooks/HOTEL_1/Original_data//search_log_room_child_hotel_1.csv', sep=';')


In [24]:
child.head(3)

id  search_log_room_id  age  baby_bed
0  1938                5797    3         0
1  1939                5915   13         0
2  1940                5915   10         0

In [25]:
len(child)

14025

In [26]:
#no duplicated rows
child[child.duplicated()]

Empty DataFrame
Columns: [id, search_log_room_id, age, baby_bed]
Index: []

In [27]:
len(child['search_log_room_id'].unique())

8685

In [28]:
len(child['id'].unique())

14025

In [29]:
child.describe()

id  search_log_room_id           age      baby_bed
count  14025.000000        14025.000000  14025.000000  14025.000000
mean    9170.879643        24382.591373      8.301390      0.055258
std     4185.930445        10633.570277      4.210544      0.228492
min     1938.000000         5797.000000      0.000000      0.000000
25%     5557.000000        14853.000000      5.000000      0.000000
50%     9143.000000        25047.000000      8.000000      0.000000
75%    12798.000000        33698.000000     11.000000      0.000000
max    16466.000000        42660.000000     17.000000      1.000000

In [30]:
child.to_csv('./../Morgens_notebooks/HOTEL_1/Clean_data/child_hotel_1.csv')

#### 4 - SEARCH tábla (search_log_hotel)
- 20.987 sor
- nincs NaN
- gond az EUR/HUF, de itt van dátum oszlop is, ide lehetne behozni az árfolyamokat először
- adattisztitás:
- dátumból: csak év-hónap-nap legyen
- conversion oszlop: NaN értékeknek -1 érték (szűréshez kell)
- total_price_final: NaN értékek nulla érték
- új oszlop: total_price_HUF !!!

In [32]:
search = pd.read_csv('./../Morgens_notebooks/HOTEL_1/Original_data/search_log_hotel_1.csv', sep=';')

In [33]:
len(search)

20987

In [34]:
len(search['search_log_session_id'].unique())

11887

In [35]:
search.head(4)

id  search_log_session_id         utc_datetime lang_code currency  \
0  112645                   4034  2024-09-17 22:19:09        hu      HUF   
1  112648                   4037  2024-09-17 22:20:15        hu      HUF   
2  112680                   4069  2024-09-17 22:27:13        de      EUR   
3  112682                   4034  2024-09-17 22:28:30        hu      HUF   

      arrival   departure  days  nights  adults  children  conversion  \
0  2024-10-21  2024-10-25     5       4       2         0         NaN   
1  2024-12-19  2024-12-22     4       3       2         1         NaN   
2  2024-09-18  2024-09-20     3       2       2         0         NaN   
3  2024-10-21  2024-10-25     5       4       4         0         NaN   

   total_price_final  
0                NaN  
1                NaN  
2                NaN  
3                NaN

In [36]:
search.sort_values(by='search_log_session_id').head(30)

id  search_log_session_id         utc_datetime lang_code currency  \
0   112645                   4034  2024-09-17 22:19:09        hu      HUF   
3   112682                   4034  2024-09-17 22:28:30        hu      HUF   
1   112648                   4037  2024-09-17 22:20:15        hu      HUF   
2   112680                   4069  2024-09-17 22:27:13        de      EUR   
4   112714                   4102  2024-09-17 22:38:42        en      EUR   
5   112757                   4145  2024-09-17 23:16:47        hu      HUF   
6   112758                   4145  2024-09-17 23:17:31        hu      HUF   
7   112759                   4146  2024-09-18 01:27:28        hu      HUF   
8   112760                   4147  2024-09-18 01:53:05        hu      HUF   
9   112761                   4147  2024-09-18 01:59:34        hu      HUF   
10  112762                   4148  2024-09-18 02:10:58        hu      HUF   
11  112764                   4150  2024-09-18 03:46:37        de      EUR   
12  112765                   4150  2024-09-18 03:47:07        de      EUR   
13  112766                   4151  2024-09-18 03:49:10        hu      HUF   
14  112767                   4151  2024-09-18 03:49:36        hu      HUF   
15  112768                   4151  2024-09-18 03:49:56        hu      HUF   
16  112770                   4153  2024-09-18 04:11:21        hu      HUF   
17  112771                   4154  2024-09-18 04:11:49        hu      HUF   
18  112772                   4155  2024-09-18 04:12:49        hu      HUF   
19  112773                   4156  2024-09-18 04:13:49        hu      HUF   
20  112774                   4156  2024-09-18 04:14:36        hu      HUF   
22  112776                   4157  2024-09-18 04:15:34        hu      HUF   
21  112775                   4157  2024-09-18 04:15:07        hu      HUF   
23  112777                   4158  2024-09-18 04:21:44        hu      HUF   
24  112778                   4159  2024-09-18 04:31:14        hu      HUF   
25  112779                   4160  2024-09-18 04:33:27        hu      HUF   
26  112780                   4160  2024-09-18 04:35:18        hu      HUF   
27  112781                   4161  2024-09-18 04:40:35        hu      HUF   
28  112782                   4162  2024-09-18 04:42:45        hu      HUF   
29  112783                   4163  2024-09-18 04:42:48        hu      HUF   

       arrival   departure  days  nights  adults  children  conversion  \
0   2024-10-21  2024-10-25     5       4       2         0         NaN   
3   2024-10-21  2024-10-25     5       4       4         0         NaN   
1   2024-12-19  2024-12-22     4       3       2         1         NaN   
2   2024-09-18  2024-09-20     3       2       2         0         NaN   
4   2024-09-25  2024-09-28     4       3       2         0         NaN   
5   2024-10-02  2024-10-06     5       4       2         0         NaN   
6   2024-10-01  2024-10-05     5       4       2         0         NaN   
7   2024-10-08  2024-10-12     5       4       2         0         NaN   
8   2024-09-18  2024-09-19     2       1       2         0         NaN   
9   2024-10-05  2024-10-06     2       1       2         0         NaN   
10  2024-10-18  2024-10-20     3       2       2         0         NaN   
11  2024-10-13  2024-10-17     5       4       2         0         NaN   
12  2024-10-13  2024-10-17     5       4       2         0         NaN   
13  2024-12-30  2024-12-31     2       1       2         3         NaN   
14  2024-12-30  2024-12-31     2       1       2         3         NaN   
15  2024-12-30  2024-12-31     2       1       2         3         NaN   
16  2024-09-27  2024-09-29     3       2       2         1         NaN   
17  2024-12-13  2024-12-17     5       4       2         0         NaN   
18  2024-12-06  2024-12-08     3       2       2         0         NaN   
19  2025-02-23  2025-02-26     4       3       2         2         NaN   
20  2025-02-23  2025-02-27     5       4       2         2        

In [37]:
search.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20987 entries, 0 to 20986
Data columns (total 13 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   id                     20987 non-null  int64  
 1   search_log_session_id  20987 non-null  int64  
 2   utc_datetime           20987 non-null  object 
 3   lang_code              20987 non-null  object 
 4   currency               20987 non-null  object 
 5   arrival                20987 non-null  object 
 6   departure              20987 non-null  object 
 7   days                   20987 non-null  int64  
 8   nights                 20987 non-null  int64  
 9   adults                 20987 non-null  int64  
 10  children               20987 non-null  int64  
 11  conversion             802 non-null    float64
 12  total_price_final      802 non-null    float64
dtypes: float64(2), int64(6), object(5)
memory usage: 2.1+ MB


In [38]:
#datatype change to date:
search['utc_datetime'] = pd.to_datetime(search['utc_datetime'])
search['arrival'] = pd.to_datetime(search['arrival'])
search['departure'] = pd.to_datetime(search['departure'])

In [39]:
#utc_datetime change to only year-month-day into a new column
search['utc_date'] = search['utc_datetime'].dt.strftime('%Y-%m-%d')

In [40]:
search.isna().sum()

id                           0
search_log_session_id        0
utc_datetime                 0
lang_code                    0
currency                     0
arrival                      0
departure                    0
days                         0
nights                       0
adults                       0
children                     0
conversion               20185
total_price_final        20185
utc_date                     0
dtype: int64

In [41]:
#Number of final bookings
20987-20185

802

In [42]:
#conversion column: replace NaN with -1
search['conversion'].fillna(-1, inplace = True)

In [43]:
#final_price column: replace NaN with 0
search['total_price_final'].fillna(0, inplace = True)

In [44]:
search.head()

id  search_log_session_id        utc_datetime lang_code currency  \
0  112645                   4034 2024-09-17 22:19:09        hu      HUF   
1  112648                   4037 2024-09-17 22:20:15        hu      HUF   
2  112680                   4069 2024-09-17 22:27:13        de      EUR   
3  112682                   4034 2024-09-17 22:28:30        hu      HUF   
4  112714                   4102 2024-09-17 22:38:42        en      EUR   

     arrival  departure  days  nights  adults  children  conversion  \
0 2024-10-21 2024-10-25     5       4       2         0        -1.0   
1 2024-12-19 2024-12-22     4       3       2         1        -1.0   
2 2024-09-18 2024-09-20     3       2       2         0        -1.0   
3 2024-10-21 2024-10-25     5       4       4         0        -1.0   
4 2024-09-25 2024-09-28     4       3       2         0        -1.0   

   total_price_final    utc_date  
0                0.0  2024-09-17  
1                0.0  2024-09-17  
2                0.0  2024-09-17  
3                0.0  2024-09-17  
4                0.0  2024-09-17

In [45]:
#no duplicated rows
search[search.duplicated()]

Empty DataFrame
Columns: [id, search_log_session_id, utc_datetime, lang_code, currency, arrival, departure, days, nights, adults, children, conversion, total_price_final, utc_date]
Index: []

In [46]:
search.to_csv('./../Morgens_notebooks/HOTEL_1/Clean_data/search_hotel_1.csv')

#### 5 - Room details (search_log_room_hotel)
- 23908 sor
- adattípusok OK
- van 22919 db NaN a picked_price, és picked_room oszlopban, ezek azért vannak mert nem választott már szobát a keresés során?
- ami nem NaN, az kód a picked_room oszlopban, legyen 0 a NaN helyett?
- ami nem NaN, az ár: HUF/EUR?, legyen 0 a NaN helyett?

In [48]:
room = pd.read_csv('./../Morgens_notebooks/HOTEL_1/Original_data/search_log_room_hotel_1.csv', sep=';')

In [49]:
room.head(15)

id  search_log_id  adults  children  picked_price picked_room
0   5794         112645       2         0           NaN         NaN
1   5797         112648       2         1           NaN         NaN
2   5829         112680       2         0           NaN         NaN
3   5831         112682       4         0           NaN         NaN
4   5863         112714       2         0           NaN         NaN
5   5906         112757       2         0           NaN         NaN
6   5907         112758       2         0           NaN         NaN
7   5908         112759       2         0           NaN         NaN
8   5909         112760       2         0           NaN         NaN
9   5910         112761       2         0           NaN         NaN
10  5911         112762       2         0           NaN         NaN
11  5913         112764       2         0           NaN         NaN
12  5914         112765       2         0           NaN         NaN
13  5915         112766       2         3           NaN         NaN
14  5916         112767       2         3           NaN         NaN

In [50]:
len(room)

23908

In [51]:
len(room['search_log_id'].unique())

20987

In [52]:
room.isna().sum()

id                   0
search_log_id        0
adults               0
children             0
picked_price     22919
picked_room      22919
dtype: int64

In [53]:
23908-22919

989

In [54]:
#no duplicated rows
room[room.duplicated()]

Empty DataFrame
Columns: [id, search_log_id, adults, children, picked_price, picked_room]
Index: []

In [55]:
room.to_csv('./../Morgens_notebooks/HOTEL_1/Clean_data/room_hotel_1.csv')

#### 6 -   ROOM_OFFER

- 109.938 sor
-  adattípusok OK
-  room price min and max can be equal! 
- room price: EUR/HUF keveredik

In [57]:
room_offer = pd.read_csv('./../Morgens_notebooks/HOTEL_1/Original_data/search_log_room_offer_hotel_1.csv', sep=";")


In [58]:
room_offer[room_offer['search_log_id'] == 112680]

id  search_log_id  search_log_room_id room_code  room_price_min  \
6  15628         112680                5829    ROOMEE          465.60   
7  15629         112680                5829    ROOMFF          504.40   
8  15630         112680                5829    ROOMGG          516.04   

   room_price_max  
6          465.60  
7          504.40  
8          516.04

In [59]:
room_offer.tail(3)

id  search_log_id  search_log_room_id room_code  room_price_min  \
109935  195320         146420               42663    ROOMEE          697.00   
109936  195321         146420               42663    ROOMFF          748.00   
109937  195322         146420               42663    ROOMHH          811.75   

        room_price_max  
109935          783.94  
109936          834.94  
109937          898.69

In [60]:
len(room_offer)

109938

In [61]:
len(room_offer['id'].unique())

109938

In [62]:
len(room_offer['search_log_id'].unique())

18528

In [63]:
len(room_offer['search_log_room_id'].unique())

20944

In [64]:
room_offer.isna().sum()

id                    0
search_log_id         0
search_log_room_id    0
room_code             0
room_price_min        0
room_price_max        0
dtype: int64

In [65]:
#no duplicated rows
room_offer[room_offer.duplicated()]

Empty DataFrame
Columns: [id, search_log_id, search_log_room_id, room_code, room_price_min, room_price_max]
Index: []

In [66]:
room_offer.to_csv('./../Morgens_notebooks/HOTEL_1/Clean_data/room_offer_hotel_1.csv')

#### 7 -  Session
- 11887 sor
- üres sorok a utm-es oszlopokban, amik nem fontosak a Morgens szerint

In [26]:
session = pd.read_csv('./../Morgens_notebooks/HOTEL_1/Original_data/search_log_session_hotel_1.csv', sep=";")

In [28]:
session.head(50)

id                                  uuid  session_id      utm_source  \
0   4034                 1145549807.1726611234  1726611549             NaN   
1   4037                 2039466988.1726611511  1726611615        facebook   
2   4069                 1556994523.1726611965  1726612033          google   
3   4102                  901475575.1726612653  1726612722          google   
4   4145                  408000469.1726614788  1726615007             NaN   
5   4146                 1097283936.1726622460  1726622848             NaN   
6   4147                 2046757301.1726624293  1726624385        facebook   
7   4148                  261439511.1726615929  1726625458  l.facebook.com   
8   4150                 1877848240.1715013016  1726631197             NaN   
9   4151                  369265528.1726631243  1726631350             NaN   
10  4153                 1386321193.1726632616  1726632681        (direct)   
11  4154                 1304073186.1726331667  1726632709          google   
12  4155                  270862531.1719673907  1726632769             NaN   
13  4156                 1312878807.1714135539  1726632829             NaN   
14  4157                 1020061150.1725873334  1726632907          google   
15  4158                 1522061108.1726546894  1726633304        facebook   
16  4159                 1533169429.1726633772  1726633874             NaN   
17  4160                  421794480.1726633853  1726634007          google   
18  4161                 1462885325.1726634368  1726634435             NaN   
19  4162                  810043846.1713843015  1726634565          google   
20  4163                 1971804983.1726634508  1726634568             NaN   
21  4164                  896673736.1726634444  1726634590             NaN   
22  4165                 1811629401.1726306412  1726634667             NaN   
23  4166                  439205377.1726634344  1726634750             NaN   
24  4167                 1524145074.1726634343  1726634850             NaN   
25  4168                 1845591774.1726634686  1726634850        (direct)   
26  4169                 2114662708.1726559733  1726635033             NaN   
27  4170                  935700975.1726597701  1726635399             NaN   
28  4171                  836310086.1714714301  1726635435             NaN   
29  4172                 1975934372.1726083991  1726635936          google   
30  4173                 1530490764.1726467263  1726636230          google   
31  4174                  368960446.1726465979  1726636302             NaN   
32  4175                  445722159.1726636245  1726636334             NaN   
33  4176                  993960798.1714038957  1726636651          google   
34  4177                 1408879783.1718880464  1726636666     www.kapu.hu   
35  4178                 1165632336.1726199126  1726636965        facebook   
36  4179                  581180701.1726637100  1726637165        facebook   
37  4180                 1779497185.1726637280  1726637349  l.facebook.com   
38  4181                 1776480123.1726637572  1726637615          google   
39  4182                   17167614.1726637360  1726637697             NaN   
40  4183  bcc7d23a-0877-4a70-9e75-8688b1dd468a  1726637897          google   
41  4184                 1486866591.1726637965  1726637998          google   
42  4185                 1058973659.1726570404  1726638011          google   
43  4186                   49306819.1714195638  1726638455             NaN   
44  4187                 1847500425.1726638272  1726638488             NaN   
45  4188                 1506793319.1722501130  1726638506             NaN   
46  4190                 1302856589.1714026790  1726638974             NaN   
47  4191                  132363178.1726639065  1726639152             NaN   
48  4192                   23952684.1724269803  1726639234       hirlevel+   
49  4193                 1187645351.1726639102  1726639383             NaN   

   utm_medium   utm_campaig

In [70]:
session.isna().sum()

id                 0
uuid               0
session_id         0
utm_source       697
utm_medium       697
utm_campaign    1555
dtype: int64

In [71]:
len(session['session_id'].unique())

11682

In [72]:
session.groupby('session_id').count()

id  uuid  utm_source  utm_medium  utm_campaign
session_id                                                
1726611549   1     1           0           0             1
1726611615   1     1           1           1             1
1726612033   1     1           1           1             0
1726612722   1     1           1           1             0
1726615007   1     1           0           0             1
...         ..   ...         ...         ...           ...
1729284680   1     1           1           1             0
1729285077   1     1           1           1             1
1729286582   1     1           1           1             1
1729286949   1     1           1           1             1
1729287722   1     1           1           1             1

[11682 rows x 5 columns]

In [73]:
session.groupby('utm_source').count().sort_values(by='id', ascending=False)

id  uuid  session_id  utm_medium  utm_campaign
utm_source                                                                    
google                        7861  7861        7861        7861          6306
(direct)                      1486  1486        1486        1486          1486
facebook                       778   778         778         778           778
bing                           268   268         268         268           268
l.facebook.com                 129   129         129         129           129
m.facebook.com                  98    98          98          98            98
redir.d-edgeconnect.media       82    82          82          82            82
ajanlat                         71    71          71          71            71
source_1                        69    69          69          69            69
ntp.msn.com                     55    55          55          55            55
seznam                          40    40          40          40            40
l.instagram.com                 34    34          34          34            34
checkout.simplepay.hu           28    28          28          28            28
lm.facebook.com                 25    25          25          25            25
hirlevel+                       20    20          20          20            20
live                            14    14          14          14            14
www.saferpay.com                13    13          13          13            13
loyalty                         12    12          12          12            12
kereso.startlap.hu              12    12          12          12            12
www.msn.com                      8     8           8           8             8
instagram.com                    8     8           8           8             8
yahoo                            7     7           7           7             7
phone                            6     6           6           6             6
hotelcms.hu                      6     6           6           6             6
source_2                         5     5           5           5             5
source_8                         5     5           5           5             5
email                            4     4           4           4             4
www.otpbankdirekt.hu             4     4           4           4             4
mail.centrum.cz                  4     4           4           4             4
hirlevel_s1                      4     4           4           4             4
source_7                         3     3           3           3             3
source_5                         3     3           3           3             3
www.kapu.hu                      2     2           2           2             2
www.glamour.hu                   2     2           2           2             2
statics.teams.cdn.office.net     2     2           2           2             2
internetbank.otpbank.hu          2     2           2           2             2
source_24                        2     2           2           2             2
hirlevel_s4                      2     2           2           2             2
search.brave.com                 2     2           2           2             2
m.hvg.hu                         1     1           1           1             1
fb                               1     1           1           1             1
gmail.hu                         1     1           1           1             1
hirlevel_s3                      1     1           1           1             1
acssv.otpbank.hu                 1     1           1           1             1
webmail.telekom.hu               1     1           1           1             1
source_10                        1     1           1           1             1
hirlevel                         1     1           1           1             1
hirlevel_s2                      1     1           1           1             1
source_6                         1     1           1           1             1
sou

In [74]:
session.groupby('utm_medium').count()

id  uuid  session_id  utm_source  utm_campaign
utm_medium                                                  
(none)      1486  1486        1486        1486          1486
cpc         7093  7093        7093        7093          5538
ct            29    29          29          29            29
email         35    35          35          35            35
offline       58    58          58          58            58
organic     1875  1875        1875        1875          1875
paid           1     1           1           1             1
referral     613   613         613         613           613

In [75]:
session.groupby('utm_medium').count()

id  uuid  session_id  utm_source  utm_campaign
utm_medium                                                  
(none)      1486  1486        1486        1486          1486
cpc         7093  7093        7093        7093          5538
ct            29    29          29          29            29
email         35    35          35          35            35
offline       58    58          58          58            58
organic     1875  1875        1875        1875          1875
paid           1     1           1           1             1
referral     613   613         613         613           613

In [76]:
#no duplicated rows
session[session.duplicated()]

Empty DataFrame
Columns: [id, uuid, session_id, utm_source, utm_medium, utm_campaign]
Index: []

In [77]:
session.to_csv('./../Morgens_notebooks/HOTEL_1/Clean_data/session_hotel_1.csv')

#### 8 -  DAILY_USERS
- 1923 sor
- date oszlop: adattípus változtatás

In [30]:
daily_users = pd.read_csv('./../Morgens_notebooks/HOTEL_1/Original_data/website_daily_users_hotel_1.csv', sep=";")

In [32]:
daily_users.head(30)

date                   utm_source_and_medium       utm_campaign  \
0   2024-09-18                               (not set)          (not set)   
1   2024-09-18                       (direct) / (none)           (direct)   
2   2024-09-18                            google / cpc          (not set)   
3   2024-09-18                        google / organic          (organic)   
4   2024-09-18                          facebook / cpc        campaign_14   
5   2024-09-18                            google / cpc           hu_brand   
6   2024-09-18                            google / cpc        campaign_15   
7   2024-09-18                            google / cpc        campaign_16   
8   2024-09-18                            google / cpc        campaign_17   
9   2024-09-18                            google / cpc      hu_pmax_brand   
10  2024-09-18                          facebook / cpc        campaign_18   
11  2024-09-18               l.facebook.com / referral         (referral)   
12  2024-09-18               m.facebook.com / referral         (referral)   
13  2024-09-18                            google / cpc        campaign_19   
14  2024-09-18                            google / cpc          (organic)   
15  2024-09-18                  ntp.msn.com / referral         (referral)   
16  2024-09-18    redir.d-edgeconnect.media / referral         (referral)   
17  2024-09-18                            google / cpc        campaign_20   
18  2024-09-18                       ajanlat / offline         campaign_2   
19  2024-09-18                            google / cpc        campaign_22   
20  2024-09-18                          facebook / cpc        campaign_21   
21  2024-09-18                            google / cpc        campaign_24   
22  2024-09-18                      msn.com / referral         (referral)   
23  2024-09-18                                source_1         (referral)   
24  2024-09-18                            google / cpc        campaign_23   
25  2024-09-18                            google / cpc        campaign_25   
26  2024-09-18                       ajanlat / offline  proposal-by-phone   
27  2024-09-18  googleads.g.doubleclick.net / referral         (referral)   
28  2024-09-18                              bing / cpc        campaign_36   
29  2024-09-18                              bing / cpc         campaign_9   

    user_count  session_count  
0          545            690  
1          253            304  
2          259            273  
3          144            170  
4          125            128  
5           97            114  
6           99            105  
7           67             76  
8           54             58  
9           41             43  
10          27             27  
11          24             26  
12          15             24  
13          15             24  
14          17             21  
15          13             17  
16          16             17  
17          13             17  
18          13             17  
19          11             14  
20          13             14  
21           9             14  
22           6             13  
23           9             11  
24          10             10  
25           7              9  
26           7              9  
27           8              8  
28           0              8  
29           8              8

In [81]:
daily_users.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1923 entries, 0 to 1922
Data columns (total 5 columns):
 #   Column                 Non-Null Count  Dtype 
---  ------                 --------------  ----- 
 0   date                   1923 non-null   object
 1   utm_source_and_medium  1923 non-null   object
 2   utm_campaign           1923 non-null   object
 3   user_count             1923 non-null   int64 
 4   session_count          1923 non-null   int64 
dtypes: int64(2), object(3)
memory usage: 75.2+ KB


In [82]:
#datatype change to date
daily_users['date'] = pd.to_datetime(daily_users['date'])


In [83]:
daily_users.isna().sum()

date                     0
utm_source_and_medium    0
utm_campaign             0
user_count               0
session_count            0
dtype: int64

In [84]:
#no duplicated rows
daily_users[daily_users.duplicated()]

Empty DataFrame
Columns: [date, utm_source_and_medium, utm_campaign, user_count, session_count]
Index: []

In [85]:
daily_users.to_csv('./../Morgens_notebooks/HOTEL_1/Clean_data/daily_users_hotel_1.csv')

#### 9 -  BUDGET
- 31 sor
- első oszlopot átnevezni
- adattípus változtatás

In [87]:
budget = pd.read_csv('./../Morgens_notebooks/HOTEL_1/Original_data/daily_ppc_budget_hotel_1.csv', sep=",")

In [88]:
budget.head()

Unnamed: 0  daily_google_spend  daily_microsoft_spend  daily_meta_spend
0  2024.09.18.              128044                  10564             35950
1  2024.09.19.              137322                  12048             53112
2  2024.09.20.              124825                   8484             61717
3  2024.09.21.              106588                   6976             64140
4  2024.09.22.              144448                   6684             73310

In [89]:
#Rename the first column
budget.rename(columns = {'Unnamed: 0':'date'}, inplace = True)

In [90]:
budget.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 31 entries, 0 to 30
Data columns (total 4 columns):
 #   Column                 Non-Null Count  Dtype 
---  ------                 --------------  ----- 
 0   date                   31 non-null     object
 1   daily_google_spend     31 non-null     int64 
 2   daily_microsoft_spend  31 non-null     int64 
 3   daily_meta_spend       31 non-null     int64 
dtypes: int64(3), object(1)
memory usage: 1.1+ KB


In [91]:
#change date type:
budget['date'] = pd.to_datetime(budget['date'])

In [92]:
budget.isna().sum()

date                     0
daily_google_spend       0
daily_microsoft_spend    0
daily_meta_spend         0
dtype: int64

In [93]:
#no duplicated rows
budget[budget.duplicated()]

Empty DataFrame
Columns: [date, daily_google_spend, daily_microsoft_spend, daily_meta_spend]
Index: []

In [94]:
budget.to_csv('./../Morgens_notebooks/HOTEL_1/Clean_data/budget_hotel_1.csv')

#### 10 -  DATEPICKERS
- 1158 sor
- date oszlop adatmódosítás

In [96]:
datepickers = pd.read_csv('./../Morgens_notebooks/HOTEL_1/Original_data/datepicker_daily_visitors_hotel_1.csv', sep=";")


In [97]:
datepickers.head()

date utm_source_and_medium utm_campaign  user_count  session_count
0  2024-09-18     (direct) / (none)     (direct)         104            124
1  2024-09-18          google / cpc    (not set)          59             59
2  2024-09-18      google / organic    (organic)          37             40
3  2024-09-18          google / cpc     hu_brand          30             34
4  2024-09-18          google / cpc  campaign_16          23             25

In [98]:
len(datepickers)

1158

In [99]:
len(datepickers['date'].unique())

31

In [100]:
len(datepickers['utm_source_and_medium'].unique())

68

In [101]:
len(datepickers['utm_campaign'].unique())

56

In [102]:
datepickers.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1158 entries, 0 to 1157
Data columns (total 5 columns):
 #   Column                 Non-Null Count  Dtype 
---  ------                 --------------  ----- 
 0   date                   1158 non-null   object
 1   utm_source_and_medium  1158 non-null   object
 2   utm_campaign           1158 non-null   object
 3   user_count             1158 non-null   int64 
 4   session_count          1158 non-null   int64 
dtypes: int64(2), object(3)
memory usage: 45.4+ KB


In [103]:
datepickers['date'] = pd.to_datetime(datepickers['date'])

In [104]:
datepickers.isna().sum()

date                     0
utm_source_and_medium    0
utm_campaign             0
user_count               0
session_count            0
dtype: int64

In [105]:
#no duplicated rows
datepickers[datepickers.duplicated()]

Empty DataFrame
Columns: [date, utm_source_and_medium, utm_campaign, user_count, session_count]
Index: []

In [106]:
datepickers.to_csv('./../Morgens_notebooks/HOTEL_1/Clean_data/datepickers_hotel_1.csv')

#### 11 - OCCUPANCY
- 10788 dor, mert 31 napon át nézzük (x) a 348 napos időszakot
- két dátum oszlopnál adattípus módosítás

In [108]:
occupancy = pd.read_csv('./../Morgens_notebooks/HOTEL_1/Original_data/daily_occupancy_hotel_1.csv', sep=";")

In [109]:
occupancy

recording_date subject_date  fill_rate
0         2024-09-18   2024-09-18     67.368
1         2024-09-18   2024-09-19     65.789
2         2024-09-18   2024-09-20     67.895
3         2024-09-18   2024-09-21     59.474
4         2024-09-18   2024-09-22     43.684
...              ...          ...        ...
10783     2024-10-18   2025-08-27      0.526
10784     2024-10-18   2025-08-28      0.526
10785     2024-10-18   2025-08-29      0.526
10786     2024-10-18   2025-08-30      0.526
10787     2024-10-18   2025-08-31      0.526

[10788 rows x 3 columns]

In [110]:
occupancy.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10788 entries, 0 to 10787
Data columns (total 3 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   recording_date  10788 non-null  object 
 1   subject_date    10788 non-null  object 
 2   fill_rate       10788 non-null  float64
dtypes: float64(1), object(2)
memory usage: 253.0+ KB


In [111]:
occupancy['recording_date'] = pd.to_datetime(occupancy['recording_date'])
occupancy['subject_date'] = pd.to_datetime(occupancy['subject_date'])

In [112]:
occupancy.isna().sum()

recording_date    0
subject_date      0
fill_rate         0
dtype: int64

In [113]:
#no duplicated rows
occupancy[occupancy.duplicated()]

Empty DataFrame
Columns: [recording_date, subject_date, fill_rate]
Index: []

In [114]:
occupancy.to_csv('./../Morgens_notebooks/HOTEL_1/Clean_data/occupancy_hotel_1.csv')